lib/chef/node/attribute_collections.rb

```diff
@@ -187,7 +187,7 @@ def convert_value(value)
         when Hash
           VividMash.new(root, value)
         when Array
-          AttrArray.new(root, value)
+          AttrArray.new(root, value.map { |e| convert_value(e) })
```

Comment:
this seems like its being patched in the wrong place and that AttrArray should be calling convert_value in the constructor.   in fact, i'm really surprised this changed between 11 and 12.   pretty sure this is the wrong way to patch this and that trying to fix it more correctly should surface what really went wrong here.